File: backend/external-api/pkg/utils/response.go
// Package utils contient les fonctions utilitaires partagées
package utils

import (
	"encoding/json"
	"external-api/internal/models"
	"net/http"
)

// WriteSuccess écrit une réponse JSON de succès standardisée
func WriteSuccess(w http.ResponseWriter, statusCode int, message string, data interface{}) {
	w.Header().Set("Content-Type", "application/json; charset=utf-8")
	w.WriteHeader(statusCode)
	json.NewEncoder(w).Encode(models.APIResponse{
		Status:  "success",
		Message: message,
		Data:    data,
	})
}

// WriteError écrit une réponse JSON d'erreur standardisée
func WriteError(w http.ResponseWriter, statusCode int, message, detail string) {
	w.Header().Set("Content-Type", "application/json; charset=utf-8")
	w.WriteHeader(statusCode)
	json.NewEncoder(w).Encode(models.APIResponse{
		Status:  "error",
		Message: message,
		Error:   detail,
	})
}

// NullableString retourne une string ou vide si nil
func NullableString(s *string) string {
	if s == nil {
		return ""
	}
	return *s
}

File: backend/external-api-php-v2/classes/Response.php
<?php

/**
 * Response — Réponses JSON standardisées
 * 
 * Toutes les réponses de l'API ont le même format :
 * { "status": "success|error", "message": "...", "data": {...} }
 */
class Response {

    /**
     * Envoie une réponse de succès JSON et arrête l'exécution
     */
    public static function success($data, string $message = "Succès", int $code = 200): void {
        self::send($code, [
            "status"  => "success",
            "message" => $message,
            "data"    => $data
        ]);
    }

    /**
     * Envoie une réponse d'erreur JSON et arrête l'exécution
     */
    public static function error(string $message, string $detail = "", int $code = 500): void {
        self::send($code, [
            "status"  => "error",
            "message" => $message,
            "error"   => $detail
        ]);
    }

    /**
     * Envoie la réponse HTTP avec les bons headers
     */
    private static function send(int $code, array $body): void {
        // Vider tout buffer de sortie parasite
        if (ob_get_length()) ob_clean();

        http_response_code($code);
        header("Content-Type: application/json; charset=utf-8");
        header("X-Content-Type-Options: nosniff");
        header("Cache-Control: no-store, no-cache");
        header("Access-Control-Allow-Origin: *");
        header("Access-Control-Allow-Methods: GET, OPTIONS");
        header("Access-Control-Allow-Headers: X-Bank-ID, X-API-Key, Content-Type");

        echo json_encode($body, JSON_UNESCAPED_UNICODE | JSON_PRETTY_PRINT);
        exit;
    }
}